subprojects/gradle/gradle-logger/src/main/kotlin/com/avito/logger/GradleLoggerFactory.kt

```diff
@@ -0,0 +1,68 @@
+package com.avito.logger
+
+import com.avito.android.elastic.ElasticConfig
+import com.avito.android.sentry.SentryConfig
+import com.avito.android.sentry.sentryConfig
+import com.avito.utils.gradle.BuildEnvironment
+import com.avito.utils.gradle.buildEnvironment
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import java.io.Serializable
+
+class GradleLoggerFactory(
+    private val isCiRun: Boolean,
+    private val sentryConfig: SentryConfig,
+    private val elasticConfig: ElasticConfig,
+    private val projectPath: String,
+    private val pluginName: String? = null,
+    private val taskName: String? = null
+) : LoggerFactory, Serializable {
+
+    override fun create(tag: String): Logger = LoggerRegistry.create(
+        isCiRun = isCiRun,
+        tag = tag,
+        projectPath = projectPath,
+        sentryConfig = sentryConfig,
+        elasticConfig = elasticConfig,
+        pluginName = pluginName,
+        taskName = taskName
+    )
+
+    companion object {
+
+        inline fun <reified T : Task> getLogger(task: T): Logger = fromTask(task).create<T>()
+
+        inline fun <reified T : Plugin<*>> getLogger(plugin: T, project: Project): Logger =
+            fromPlugin(plugin, project).create<T>()
+
+        fun fromTask(task: Task): GradleLoggerFactory = fromProject(
+            project = task.project,
+            taskName = task.name
+        )
+
+        fun fromPlugin(
+            plugin: Plugin<*>,
+            project: Project
+        ): GradleLoggerFactory = fromProject(
+            project = project,
+            pluginName = plugin.javaClass.simpleName
+        )
+
+        fun fromProject(
```

Comment:
When should we use this function?
It seems internal/private because of optional `pluginName` and `taskName`. If I have any of them I should use the corresponding functions. If I'm not in plugin or task I don't need to pass these parameters.

Reply:
https://github.com/avito-tech/avito-android/pull/695/files#diff-2bdb0d761ed206659ed066b22bf76177f7aa5630bccb88f6e1809647d5a587f4R48

for example, we lost info about task but have project; maybe it's worth refactor, but i see no harm in this fun stays public